src/main/java/org/opensearch/geospatial/ip2geo/common/DatasourceState.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geospatial.ip2geo.common;
+
+/**
+ * Ip2Geo datasource state
+ *
+ * When data source is created, it starts with PREPARING state. Once the first GeoIP data is generated, the state changes to AVAILABLE.
+ * Only when the first GeoIP data generation failed, the state changes to FAILED.
+ * Subsequent GeoIP data failure won't change data source state from AVAILABLE to FAILED.
+ * When delete request is received, the data source state changes to DELETING.
+ *
+ * State changed from left to right for the entire lifecycle of a datasource
+ * (PREPARING) to (FAILED or AVAILABLE) to (DELETING)
+ *
+ */
+public enum DatasourceState {
```

Comment:
should we have an expired state also? which will make sure that if we are not able to delete the data source it is still marked not for use.

Reply:
Expired or not will be decided based on the last updated date and valid duration for the data.